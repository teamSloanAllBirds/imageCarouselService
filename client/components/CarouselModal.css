.ReactModal__Overlay {
  opacity: 1;
  /* transform: translateY(100px); */
  /* transition: all 300ms ease-in-out; */
  animation: bounce .6s;
}

.ReactModal__Overlay--after-open {
  background: rgba(0, 0, 0, 0.7);
  opacity: 1;
  transform: translateX(0px);
}

@keyframes bounce {
  0% {
    transform: translateY(35%);
    /* opacity: 0; */
    /* animation-timing-function: ease-out; */
  }
  50% {
    opacity: 1;
    transform: translateY(0%);
    /* animation-timing-function: ease-in; */
  }
  75% {
    transform: translateY(1%);
    animation-timing-function: ease-in;
  }
  100% {
    transform: translateY(0%);
    animation-timing-function: ease-in;
  }
}

.ReactModal__Overlay--before-close {
  opacity: 1;
  transform: translateX(-100px);
}

.modal {
  padding: 5px;
}

.xicon {
  cursor: pointer;
  opacity: .7;
}

.xiconspin {
  cursor: pointer;
  opacity: .7;
  animation: spin .3s;
}

.xiconunspin {
  cursor: pointer;
  opacity: .7;
  animation: unspin .3s;
}

@keyframes spin {
  from {
    transform:rotate(0deg);
  }
  to {
    transform:rotate(90deg);
  }
}

@keyframes unspin {
  from {
    transform:rotate(0deg);
  }
  to {
    transform:rotate(-90deg);
  }
}