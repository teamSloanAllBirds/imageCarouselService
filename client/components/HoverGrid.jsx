/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable react/no-array-index-key */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import styled from 'styled-components';

import { Icon } from '@iconify/react';
import plusCircle from '@iconify/icons-mdi/plus-circle';

const Table = styled.table`
`;

const Row = styled.tr`
`;

const Td = styled.td`
  padding: 5px;
  .container {
    position:relative;
    text-align: center;
    vertical-align: middle;
  }

  img {
    width: 30vw;
    max-width: 350px;
    height: 30vw;
    max-height: 350px;
  }

  img:hover, .icon:hover {
    cursor: pointer;
  }

  .icon{
    position: absolute;
    display:none;
    color: white;
    width: 40px;
    height: 40px;
    z-index:100;
  }

  div:hover .icon {
    display: block;
    opacity: .5;
    top: 45%;
    left: 45%;
  }

  div: hover .icon:hover {
    opacity: 1;
  }
`;

class HoverGrid extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.clickHandler = this.clickHandler.bind(this);
  }

  clickHandler(e) {
    const { selectImage } = this.props;
    selectImage(parseInt(e.target.dataset.index, 10));
  }

  render() {
    const { urls, toggleModal } = this.props;
    const tds = urls.map((url, index) => (
      <Td key={index} onClick={toggleModal}>
        <a href={`#slide-${index}`}>
          <div className="container">
            <img alt="pic" src={url} data-index={index} onClick={this.clickHandler} />
            <Icon icon={plusCircle} className="icon" />
          </div>
        </a>
      </Td>
    ));
    const pairs = tds.reduce((r, v, i) => {
      if (i % 2 === 0) { r.push(tds.slice(i, i + 2)); }
      return r;
    }, []);
    const rows = pairs.map((pair, index) => (
      <Row key={index}>
        {pair[0]}
        {pair[1]}
      </Row>
    ));
    return (
      <div>
        <Table>
          <tbody>
            {rows}
          </tbody>
        </Table>
      </div>
    );
  }
}

export default HoverGrid;
