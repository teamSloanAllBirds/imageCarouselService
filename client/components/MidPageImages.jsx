/* eslint-disable import/extensions */
/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components';

const MidTable = styled.table`
  border-spacing: 0 5vw;
  width: 100%
`;

const RowTable = styled.table`
  height: 100%
`;

const BiggerDiv = styled.div`
  overflow: hidden;
  border: 1px solid red;
  max-height: 300px;
  width: 40vw;
  max-width: 400px;
`;

const SmallerDiv = styled.div`
  border: 1px solid blue;
  height: 100%;
  width: 35vw;
  max-width: 350px;
  `;

const TextContainer = styled.div`
  font-family: 'Meticula';
  padding: 20%;
`;

const ImageContainer = styled.div`
  img{
    display:block;
    height: 200%;
    width: 200%;
    margin: -50% -50%;
    transition: transform .5s ease;
  }
  img:hover{
    transform: scale(1.05);
  }
`;

function MidPageImages({ urls }) {
  const rows = urls.map((url, index) => {
    if (index % 2 === 0) {
      return (
        <tr key={url}>
          <td>
            <RowTable>
              <tbody>
                <tr>
                  <td>
                    <SmallerDiv>
                      <TextContainer>
                        This is the sample text in a font
                      </TextContainer>
                    </SmallerDiv>
                  </td>
                  <td>
                    <BiggerDiv>
                      <ImageContainer>
                        <img alt="sneakerpic" src={url} width="400px" height="350px" />
                      </ImageContainer>
                    </BiggerDiv>
                  </td>
                </tr>
              </tbody>
            </RowTable>
          </td>
        </tr>
      );
    }
    return (
      <tr key={url}>
        <td>
          <RowTable>
            <tbody>
              <tr>
                <td>
                  <BiggerDiv>
                    <ImageContainer>
                      <img alt="sneakerpic" src={url} width="400px" height="350px" />
                    </ImageContainer>
                  </BiggerDiv>
                </td>
                <td>
                  <SmallerDiv>
                    <TextContainer>
                      This is the sample text in a font
                    </TextContainer>
                  </SmallerDiv>
                </td>
              </tr>
            </tbody>
          </RowTable>
        </td>
      </tr>
    );
  });

  return (
    <div>
      <MidTable>
        <tbody>
          {rows}
        </tbody>
      </MidTable>
    </div>
  );
}

export default MidPageImages;
