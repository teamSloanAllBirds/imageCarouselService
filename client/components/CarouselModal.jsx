/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable no-nested-ternary */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import Modal from 'react-modal';
import { Icon, InlineIcon } from '@iconify/react';
import xIcon from '@iconify/icons-bi/x';
import bxsChevronRight from '@iconify/icons-bx/bxs-chevron-right';
import bxsChevronLeft from '@iconify/icons-bx/bxs-chevron-left';
import dotIcon from '@iconify/icons-bi/dot';
import './CarouselModal.css';

Modal.setAppElement(document.getElementById('app'));

class CarouselModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      xHover: false,
      counter: 0,
      modalMeasurement: '200px',
    };
    this.hoverHandler = this.hoverHandler.bind(this);
    this.updateDimensions = this.updateDimensions.bind(this);
    this.nextImage = this.nextImage.bind(this);
    this.prevImage = this.prevImage.bind(this);
  }

  componentDidMount() {
    this.updateDimensions();
    window.addEventListener('resize', this.updateDimensions);
  }

  updateDimensions() {
    this.setState({ modalMeasurement: `${window.innerWidth < window.innerHeight ? 0.9 * window.innerWidth : 0.9 * window.innerHeight}px` });
  }

  hoverHandler() {
    const { xHover, counter } = this.state;
    this.setState({ xHover: !xHover, counter: counter + 1 });
  }

  nextImage() {
    const { urls, currentIndex, selectImage } = this.props;
    selectImage(urls[currentIndex === urls.length - 1 ? 0 : currentIndex + 1]);
  }

  prevImage() {
    const { urls, currentIndex, selectImage } = this.props;
    selectImage(urls[currentIndex === 0 ? urls.length - 1 : currentIndex - 1]);
  }

  render() {
    const {
      urls,
      modal,
      current,
      currentIndex,
      toggleModal,
    } = this.props;
    const { xHover, counter, modalMeasurement } = this.state;
    const dots = urls.map((c, index) => <InlineIcon icon={dotIcon} key={c} className="dot" color="gray" height={index === currentIndex ? '40' : '20'} width={index === currentIndex ? '40' : '20'} />);
    return (
      <div id="container">
        <Modal
          isOpen={modal}
          onRequestClose={toggleModal}
          style={{
            overlay: {
              position: 'fixed',
              top: 0,
              left: 0,
              right: 0,
              bottom: 0,
              backgroundColor: 'rgba(0, 0, 0, 0.75)',
            },
            content: {
              top: '5%',
              bottom: '5%',
              height: modalMeasurement,
              width: modalMeasurement,
              padding: '0 0 0 0',
              border: '0px solid #ccc',
              background: '#fff',
              overflow: 'auto',
              WebkitOverflowScrolling: 'touch',
              borderRadius: '0px',
              outline: 'none',
            },
          }}
        >
          <div className="modal">
            <Icon icon={xIcon} onMouseEnter={this.hoverHandler} onMouseLeave={this.hoverHandler} onClick={toggleModal} className={!xHover && counter === 0 ? null : xHover ? 'xiconspin' : 'xiconunspin'} id="icon" width="40" height="40" />
            <div id="leftArrowDiv" onClick={this.prevImage}>
              <Icon icon={bxsChevronLeft} className="leftChevron" id="icon" width="20" height="20" />
            </div>
            <div id="rightArrowDiv" onClick={this.nextImage}>
              <Icon icon={bxsChevronRight} className="rightChevron" id="icon" width="20" height="20" />
            </div>
            <div id="carouselDots">
              {dots}
            </div>
            <img id="current" alt="current" src={current} width="100%" height="100%" />
          </div>
        </Modal>
      </div>
    );
  }
}

export default CarouselModal;
