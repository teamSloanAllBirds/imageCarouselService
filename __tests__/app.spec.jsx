import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/app.jsx';
import HoverGrid from '../client/components/HoverGrid.jsx';
import CarouselModal from '../client/components/CarouselModal.jsx';
import MidPageImages from '../client/components/MidPageImages.jsx';

configure({ adapter: new Adapter() });

const urls =[
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/001.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/002.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/003.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/004.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/005.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/006.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/007.jpeg',
  'https://teamsloanpics.s3.us-east-2.amazonaws.com/008.jpeg'
];


describe('App testing', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />);
    wrapper.setState({
      urls: urls,
      modal: false,
      current: '',
      currentIndex: 0,
    });
  });

  it('should render components', () => {
    expect(wrapper.find(HoverGrid)).toHaveLength(1);
    expect(wrapper.find(CarouselModal)).toHaveLength(1);
    expect(wrapper.find(MidPageImages)).toHaveLength(1);
  });
});

describe('HoverGrid testing', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<HoverGrid urls={ urls }/>);
  });

  it('displays all images passed as props', () => {
    expect(wrapper.find('div.container')).toHaveLength(urls.length);
  });

  it('makes all images clickable', () => {
    const images = wrapper.find('.container');
    images.forEach((image) => expect(() => { image.simulate('click'); }).not.toThrow(Error));
  });
});

describe('CarouselModal testing', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<CarouselModal urls={ urls }/>);
  });

  it('does not display when modal is false', () => {
    wrapper.setProps({modal: false});
    expect(wrapper.find('div.modal')).toHaveLength(0);
  });
});

describe('MidPageImages testing', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<MidPageImages urls={ urls }/>);
  });

  it('displays all images passed as props', () => {
    expect(wrapper.find('img')).toHaveLength(urls.length);
  });
});