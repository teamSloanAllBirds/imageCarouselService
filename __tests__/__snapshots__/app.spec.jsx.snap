// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App functionality should render and mount properly 1`] = `ReactWrapper {}`;

exports[`App functionality should render without throwing an error 1`] = `ShallowWrapper {}`;

exports[`test HoverGrid 1`] = `
<div>
  <table
    className="sc-AxjAm fDEmJb"
  >
    <tbody>
      <tr
        className="sc-AxirZ dqVvws"
      >
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/001.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/002.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
      </tr>
      <tr
        className="sc-AxirZ dqVvws"
      >
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/003.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/004.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
      </tr>
      <tr
        className="sc-AxirZ dqVvws"
      >
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/005.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/006.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
      </tr>
      <tr
        className="sc-AxirZ dqVvws"
      >
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/007.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
        <td
          className="sc-AxiKw edaVZL"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <img
              alt="pic"
              onClick={[Function]}
              src="https://teamsloanpics.s3.us-east-2.amazonaws.com/008.jpeg"
            />
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M17 13h-4v4h-2v-4H7v-2h4V7h2v4h4m-5-9A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
